doc/src/explanation/solve_output.rst

```diff
@@ -0,0 +1,120 @@
+solutions as dictionaries or ``set=True`` to get a tuple containing the list
+of variables and the set of values for those variables. (These settings only
+affect the output when the input does not contain relational expressions.)
+
+    >>> from sympy import sqrt, exp, solve, Symbol, Eq
+    >>> from sympy.abc import x, y, z, a, b
+    >>> solve(x, x, dict=True)
+    [{x: 0}]
+    >>> solve([x**2 - y, x + y - 6], set=True)
+    ([x, y], {(-3, 9), (2, 4)})
+
+The following is an explanation about the conditions under which various
+forms of output that are obtained by *default*.
+
+1. An empty list (or set) indicates that no solution was found.
+
+    >>> [] == solve(sqrt(x) + 1) == solve(sqrt(x) + 1, dict=True)
+    True
+    >>> solve(sqrt(x) + 1, set=True)
+    ([x], set())
+
+2. A list of values is given when the symbol to solve for was
+unambiguous in context because a) the equation was univariate and
+no symbols were specified or b) a single symbol was specified as
+being of interest.
+
+    >>> solve(x**2 - 4)
+    [-2, 2]
+    >>> solve(x - y - 1, x)
+    [y + 1]
+
+3. A single dictionary with keys being symbols and values being the solutions
+for those symbols is the result when equations are passed as a list and are
+all linear in the symbols given. Such a system is automatically generated
+for a single equation (not passed as a list) if there is an undetermined
+coefficients solution for the symbols specified.
+
+    >>> solve([x + y - 2, x - y + 2], x, y)
+    {x: 0, y: 2}
+    >>> solve(a*x - 2*x + b - 5, {a, b})  # undetermined coefficients
+    {a: 2, b: 5}
+
+4. A list of tuples, each giving a solution for the symbols in the order
+they were given, is given when a) a list of equations containing at least
+one nonlinear equation or b) a list of symbols is given in a well defined
+order. (This is also the format for the tuples in the set returned when
+using the flag ``set=True``.)
+
+    >>> solve(x - 1, x, y)  # more than one symbol
+    [(1, y)]
+    >>> solve([x**2], x)  # list with nonlinear equation
+    [(0,)]
+    >>> solve([x**2 - 1], x)
+    [(-1,), (1,)]
+    >>> solve([x**2 - y, x - 3], x, y)  # nonlinear and multiple symbols
+    [(3, 9)]
+
+5. A list of dictionaries is returned when the expression was not univariate or
+there was a nonlinear expression in a list *and* the order of
+symbols would otherwise be ambiguous because a) no symbols were passed or b) the
+symbols were passed as a set. (This is also the format selected with ``dict=True``.)
+
+    >>> solve(x - y)
+    [{x: y}]
+    >>> solve([exp(x) - 1, x**2])
+    [{x: 0}]
+    >>> system = [x + y - z, x**2 - y + z]
+    >>> sol = solve(system, {x, y, z}); sol
+    [{x: -1, y: z + 1}, {x: 0, y: z}]
+
+    The dictionary will only contain values that are distinct from the keys.
+    In the last example above, there is no key for ``z`` in the dictionary
+    since the 2 equations did not provide enough constraints to determine its
```

Comment:
Typical style guides recommend spelling out numbers less than 10, so 2 -> two. In a math-focused page, it's particularly helpful to reserve numbers for the math content.